src/main/java/marquez/service/JobService.java

```diff
@@ -82,12 +89,17 @@ public Job createOrUpdate(
       if (!exists(namespaceName, jobName)) {
         final NamespaceRow namespaceRow = namespaceDao.findBy(namespaceName.getValue()).get();
         final JobRow newJobRow = Mapper.toJobRow(namespaceRow.getUuid(), jobName, meta);
-
         jobDao.insert(newJobRow);
       }
 
       final UUID version = meta.version(namespaceName, jobName);
       if (!versionDao.exists(version)) {
+        final String checksum = Utils.checksumFor(meta.getContext());
+        if (!contextDao.exists(checksum)) {
```

Comment:
are we saying if the checksum is the same and job version is different, different job version share the same job context ?  I thought each job version will come with a new job context.